test/integration/builder/additional.js

```diff
@@ -961,32 +919,27 @@ module.exports = function(knex) {
       const queryTimeout = 10;
       const secondQueryTimeout = 11;
 
-      return getTestQuery()
-        .timeout(queryTimeout, { cancel: true })
-        .then(function() {
-          throw new Error("Shouldn't have gotten here.");
-        })
-        .catch(function(error) {
-          expect(_.pick(error, 'timeout', 'name', 'message')).to.deep.equal({
-            timeout: queryTimeout,
-            name: 'error',
-            message: `After query timeout of ${queryTimeout}ms exceeded, cancelling of query failed.`,
-          });
-        })
-        .then(() => {
-          knexPrototype._wrappedCancelQueryCall = originalWrappedCancelQueryCall;
-          return getTestQuery().timeout(secondQueryTimeout, { cancel: true });
-        })
-        .catch(function(error) {
-          expect(_.pick(error, 'timeout', 'name', 'message')).to.deep.equal({
-            timeout: secondQueryTimeout,
-            name: 'KnexTimeoutError',
-            message: `Defined query timeout of ${secondQueryTimeout}ms exceeded when running query.`,
-          });
-        })
-        .finally(() => {
-          return knexDb.destroy();
+      try {
+        await expect(
+          getTestQuery().timeout(queryTimeout, { cancel: true })
+        ).to.be.eventually.rejected.and.deep.include({
+          timeout: queryTimeout,
+          name: 'error',
+          message: `After query timeout of ${queryTimeout}ms exceeded, cancelling of query failed.`,
         });
+
+        knexPrototype._wrappedCancelQueryCall = originalWrappedCancelQueryCall;
```

Comment:
This relates to my "use `sinon` sandboxing" comment from a moment ago.

Notice: if the previous `expect(..)` statement fails, this this line will never be executed.  This means that the prototype will never be restored, which will lead to some really confusing errors in later test cases.